docs/source/pytest.rst

```diff
@@ -0,0 +1,435 @@
+Example with a custom ``Makefile`` that is defining the ``sample_module`` make recipe:
+
+.. code:: python
+
+    import pytest
+    import subprocess
+
+    @pytest.mark.cocotb
+    def test_sample_module() -> None:
+        """Build and run HDL design."""
+        subprocess.run(["make", "sample_module"], check=True)
+
+
+You may also consider to use command line arguments from pytest to configure build system:
+
+.. code:: python
+
+    import pytest
+    import subprocess
+
+    @pytest.mark.cocotb
+    def test_sample_module(request: pytest.FixtureRequest) -> None:
+        args: list[str] = ["make", "sample_module"]
+
+        # Add additional arguments based on request.config.option.cocotb_* options or
+        # from fixtures request.node.iter_markers() like "cocotb" marker
+        if request.config.option.cocotb_verbose:
+            args.append("VERBOSE=1")
+
+        subprocess.run(args, check=True)
+
+
+And make it reusable for other projects/teams by packaging is as a new plugin for pytest:
+
+.. code:: python
+
+    import pytest
+    import subprocess
+
+    class MyBuildSystem:
+        def __init__(self, request: pytest.FixtureRequest) -> None:
+            # Handle request.config.option.cocotb_* options and fixtures request.node.iter_markers() like "cocotb" marker
+            self.args: list[str] = ["make"]
+
+        def build_and_run(self) -> None:
+            # Compile HDL design and run simulator
+            subprocess.run(args, check=True)
+
+
+    @pytest.fixture
+    def my_build_system(request: pytest.FixtureRequest) -> MyBuildSystem:
+        return MyBuildSystem(request)
+
+
+So others can use it in their projects:
+
+.. code:: python
+
+    import pytest
+    from pytest_cocotb_my_build_system import MyBuildSystem
+
+    @pytest.mark.cocotb
+    def test_sample_module(my_build_system: MyBuildSystem) -> None:
+        my_build_system.build_and_run()
+
+
+Configuration
+=============
+
+Thanks to :py:mod:`cocotb_tools.pytest.plugin`, cocotb can be configured in many ways.
+
+Precedence order of configuring cocotb from the highest to the lowest priority:
+
+1. :py:func:`cocotb_tools.pytest.hdl.HDL` attributes set at fixture or test function level
+2. :py:deco:`pytest.mark.cocotb` marker used with test functions.
+3. ``--cocotb-*`` command line arguments when invoking them with `pytest`_ command line interface.
+4. ``COCOTB_*`` environment variables.
+5. ``cocotb_*`` entries defined in various configuration files like `pyproject.toml`_ file.
+6. Default values.
+
+All available command line arguments, configuration entries and environment variables that can be
+used to configure cocotb testing environment, can be listed by invoking `pytest`_ help:
+
+.. code:: shell
+
+    pytest --help
+
+
+.. _pytest-plugin-options:
+
+Options
+=======
+
+.. argparse::
+   :module: cocotb_tools.pytest.plugin
+   :func: options_for_documentation
```

Comment:
`:prog:` option is required https://sphinx-argparse.readthedocs.io/en/latest/usage.html#basic-usage
```suggestion
.. argparse::
   :module: cocotb_tools.pytest.plugin
   :func: options_for_documentation
   :prog: pytest
```